packages/phasor/src/renderer.ts

```diff
@@ -8,32 +11,48 @@ export class Renderer {
   ctx: CanvasRenderingContext2D;
   gridManager = new GridManager();
 
-  readonly width: number;
-  readonly height: number;
+  private _width = 0;
+  private _height = 0;
 
   private _zoom = 1.0;
   private _centerX = 0.0;
   private _centerY = 0.0;
   private _shouldUpdate = false;
 
-  constructor(canvas: HTMLCanvasElement) {
-    const dpr = window.devicePixelRatio;
-
-    canvas.style.width = '100%';
-    canvas.style.height = '100%';
-
-    const bbox = canvas.getBoundingClientRect();
-    canvas.width = Math.ceil(bbox.width * dpr);
-    canvas.height = Math.ceil(bbox.height * dpr);
+  private _disposeResizeListener!: Disposable;
 
+  constructor(canvas: HTMLCanvasElement) {
     this.canvas = canvas;
     this.ctx = canvas.getContext('2d') as CanvasRenderingContext2D;
-    this.width = bbox.width;
-    this.height = bbox.height;
-    this._centerX = this.width / 2;
-    this._centerY = this.height / 2;
+
+    this._initSize();
+    this.setCenter(this.width / 2, this.height / 2);
 
     this._loop();
+
+    this._disposeResizeListener = Signal.disposableListener(
```

Comment:
Is it possible to listen on the resizing of the canvas element itself? This could be more precise if possible.

Reply:
I want to use window resizing event handing devicePixelRatio changing. But It seems also having some issues. I will change it to ResizeObserver.